packages/application-shell/src/components/authenticated/auth-callback.tsx

```diff
@@ -0,0 +1,113 @@
+import type { ApplicationWindow } from '@commercetools-frontend/constants';
+import type { TAsyncLocaleDataProps } from '@commercetools-frontend/i18n';
+import type { AuthorizeSessionState } from './types';
+
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { decode } from 'qss';
+import jwtDecode from 'jwt-decode';
+import { STORAGE_KEYS, LOGIN_STRATEGY_OIDC } from '../../constants';
+import Redirector from '../redirector';
+import AuthCallbackErrorPage from './auth-callback-error-page';
+
+declare let window: ApplicationWindow;
+
+export type TProps = {
+  locale: string;
+  applicationMessages: TAsyncLocaleDataProps['applicationMessages'];
+  children?: never;
+};
+type AuthorizeCallbackFragments = { sessionToken?: string; state: string };
+type SessionToken = { nonce: string };
+
+const removeSSOSessionState = (key: string) =>
+  window.sessionStorage.removeItem(key);
+const getSSOSessionState = (key: string): AuthorizeSessionState | null => {
+  const unparsedSessionState = window.sessionStorage.getItem(key);
+  if (unparsedSessionState) {
+    try {
+      const parsedSessionState = JSON.parse(
+        unparsedSessionState
+      ) as AuthorizeSessionState;
+
+      removeSSOSessionState(key);
+
+      return parsedSessionState;
+    } catch (error) {
+      removeSSOSessionState(key);
+
+      if (process.env.NODE_ENV !== 'production') {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `Cannot parse session state for "${key}".\n${unparsedSessionState}`
+        );
+      }
+    }
+  }
+  return null;
+};
+
+const AuthCallback = (props: TProps) => {
```

Comment:
If you're familiar with the MC SSO callback component, this is very similar: we parse the token from the hash fragment, decode it, validate the nonce, etc.